forge/forge.js

```diff
@@ -153,14 +153,70 @@ module.exports = async (options = {}) => {
             secret: server.settings.get('cookieSecret')
         })
         await server.register(csrf, { cookieOpts: { _signed: true, _httpOnly: true } })
+
+        let contentSecurityPolicy = false
+        if (runtimeConfig.content_security_policy?.enabled) {
+            if (!runtimeConfig.content_security_policy.directives) {
+                contentSecurityPolicy = {
+                    directives: {
+                        'base-uri': ["'self'"],
+                        'default-src': ["'self'"],
+                        'script-src': ["'self'", "'unsafe-inline'", "'unsafe-eval'"],
+                        'img-src': ["'self'", 'data:', 'www.gravatar.com'],
+                        'font-src': ["'self'"],
+                        'style-src': ["'self'", 'https:', "'unsafe-inline'"],
+                        'upgrade-insecure-requests': null
+                    }
+                }
+            } else {
+                contentSecurityPolicy = {
+                    directives: runtimeConfig.content_security_policy.directives
+                }
+            }
+
+            if (runtimeConfig.content_security_policy.report_only) {
+                contentSecurityPolicy.reportOnly = true
+                if (runtimeConfig.content_security_policy.report_uri) {
+                    contentSecurityPolicy.directives['report-uri'] = runtimeConfig.content_security_policy.report_uri
+                }
+            }
+
+            if (runtimeConfig.telemetry.frontend?.plausible?.domain) {
+                contentSecurityPolicy.directives['script-src'].push('plausible.io')
```

Comment:
Only slight concern is this assumes `directives['script-src']` exists and is an array. That'll be the case if the config hasn't provided a custom `directives` object and we've set it up ourselves (line 159 above). But if a custom object has been provided but doesn't contain everything we expect, this will fail.

Reply:
I'll add some tests that `script-src` exists and if not add it with the values (but if it doesn't exist in the user provided values it's going to break everything anyway).